src/ios/WebviewProxy.m

```diff
@@ -103,19 +118,116 @@ - (void) stopSchemeTask: (id <WKURLSchemeTask>)urlSchemeTask {
+        if ([options objectForKey:@"name"] != nil) [cookieProperties setObject:[options objectForKey:@"name"] forKey:NSHTTPCookieName];  // required
+        if ([options objectForKey:@"domain"] != nil) [cookieProperties setObject:[options objectForKey:@"domain"] forKey:NSHTTPCookieOriginURL];  // reuse required domain
+        if ([options objectForKey:@"path"] != nil) [cookieProperties setObject:[options objectForKey:@"path"] forKey:NSHTTPCookiePath]; // required
+        if ([options objectForKey:@"port"] != nil) [cookieProperties setObject:[options objectForKey:@"port"] forKey:NSHTTPCookiePort];
+        if (@available(iOS 13.0, *)) {
+            if ([options objectForKey:@"sameSitePolicy"] != nil) [cookieProperties setObject:[options objectForKey:@"sameSitePolicy"] forKey:NSHTTPCookieSameSitePolicy];
+        }
+        if ([options objectForKey:@"secure"] != nil) [cookieProperties setObject:[options objectForKey:@"secure"] forKey:NSHTTPCookieSecure];
+        if ([options objectForKey:@"value"] != nil) [cookieProperties setObject:[options objectForKey:@"value"] forKey:NSHTTPCookieValue];  // required
+        if ([options objectForKey:@"version"] != nil) [cookieProperties setObject:[options objectForKey:@"version"] forKey:NSHTTPCookieVersion];
+        
+        NSHTTPCookie *cookie = [NSHTTPCookie cookieWithProperties:cookieProperties];
+        NSLog(@"setCookie(): %@:%@:%@", cookie.domain, cookie.name, cookie.value);
+        NSHTTPCookieStorage *storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+        [storage setCookie:cookie];
+        
+        CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+    }
+    @catch (NSException *exception) {
+        NSLog(@"WebViewProxy setCookie() exception: %@", exception.debugDescription);
+        
+        CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:exception.reason];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+    }
+}
 
+- (void) clearCookies:(CDVInvokedUrlCommand*)command {
+    NSHTTPCookieStorage *storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+    
+    
+    if (storage.cookies.count > 0) {
+        for (NSHTTPCookie *cookie in storage.cookies) {
+            NSLog(@"clearCookies(): removed cookie %@:%@:%@", cookie.domain, cookie.name, cookie.value);
+            [storage deleteCookie:cookie];
+        }
+        NSLog(@"clearCookies(): all cookies cleared");
+        
+    } else {
+        NSLog(@"clearCookies(): no cookies found to be cleared");
+    }
+    
+    
+    CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
     [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+    
+//        WKWebsiteDataStore* dataStore = [WKWebsiteDataStore defaultDataStore];
```

Comment:
This fixes the callback async issues. Now it waits for the `dispatch_group` to complete before returning the response to the client.

Regardless this did not work at all as some cookies were still being kept, I had to use `NSHTTPCookieStorage` to fix that.

I must be missing something, I'd love to get some feedback on this.